### PR TITLE
add map definition index support

### PR DESCRIPTION
**Pull request by arthurdead**
so i can use m_nNextMapVoteOptions on the gamerules
no windows signature imagine using windows

image

### CODE REVIEW

**Comment by nosoop** [comment]
Nice, nice.  Not sure how much use this kind of thing would get, but it looks like something that wouldn't be too difficult to maintain upstream.  (`TF2Econ_GetItemSchemaAddress` is available for personal uses / quick prototyping, in case you missed it.)

I'll push some minor changes to the docs and get a Windows sig into the PR before I merge.  `MapDef_t` is the same across both platforms so the raw read shouldn't be a problem.

File: scripting/include/tf_econ_data.inc [hunk 1]
```diff
@@ -292,6 +292,12 @@ native ArrayList TF2Econ_GetParticleAttributeList(
  */
 native ArrayList TF2Econ_GetPaintKitDefinitionList();
 
+/**
+ * Return the definition index of the specific map.
+ * -1 if not valid.
+ */
+native int TF2Econ_GetMapDefinitionIndex(const char[] name);
```
Comment (nosoop) [inline scripting/include/tf_econ_data.inc:299]: Intending to change the native name to `TF2Econ_TranslateMapNameToDefinitionIndex` to be more in line with the other natives that go between names / values.

I could go with `TF2Econ_GetMapDefinitionIndexFromName` or `TF2Econ_GetMapDefinitionIndexByName` as well; not particularly attached to any of the three names.  The last one would match what the game uses at least.

**Review by nosoop — COMMENTED** [review]
Would like your input on these things.  No need to push any more commits for them; I'll handle it on my end.

File: scripting/tf_econ_data.sp [hunk 2]
```diff
@@ -419,6 +431,22 @@ Address GetEconAttributeDefinitionByName(const char[] name) {
 			SDKCall(g_SDKCallSchemaGetAttributeDefinitionByName, pSchema, name) : Address_Null;
 }
 
+int GetMapDefinitionIndex(const char[] name) {
+	Address pSchema = GetEconItemSchema();
+	if(!pSchema) {
+		return -1;
```
[the remark this answers is not in the document]
Comment (arthurdead) [reply scripting/tf_econ_data.sp]: thats fine i just defaulted to the old programmer habit of returning -1

File: scripting/include/tf_econ_data.inc (hunk 1 above)
Comment (arthurdead) [reply scripting/include/tf_econ_data.inc:299]: i really have no opinion on the function name it can be whatever you want

**Comment by arthurdead** [comment]
> TF2Econ_GetItemSchemaAddress` is available for personal uses / quick prototyping, in case you missed it.

i did miss it.....

**Comment by nosoop** [comment]
Alright; finished up the changes.  `TF2Econ_GetMapDefinitionIndexByName` is the finalized native name, and it returns 0 on invalid names (so it points to the "Missing" entry).

Thank you for the contribution!

**Pull request merged**